Makefile.dist-packaging

```diff
@@ -0,0 +1,35 @@
+# -*- mode: Makefile -*-
+
+GITREV = $$(git describe --always --tags)
+GITREV_FOR_PKG = $(shell echo "$(GITREV)" | sed -e 's,-,\.,g' -e 's,^v,,')
+
+srcdir=$(shell pwd)
+PACKAGE=atomic-pkglayer
+
+PKG_VER = $(PACKAGE)-$(GITREV_FOR_PKG)
+
+dist-snapshot:
+	set -x; \
+	echo "PACKAGE=$(PACKAGE)"; \
+	TARFILE_TMP=$(PKG_VER).tar.tmp; \
+	echo "Archiving $(PACKAGE) at $(GITREV)"; \
+	(cd $(srcdir); git archive --format=tar --prefix=$(PKG_VER)/ $(GITREV)) > $${TARFILE_TMP}; \
+	(cd $$(git rev-parse --show-toplevel); git submodule status) | while read line; do \
+	  rev=$$(echo $$line | cut -f 1 -d ' '); path=$$(echo $$line | cut -f 2 -d ' '); \
+	  echo "Archiving $${path} at $${rev}"; \
+	  (cd $(srcdir)/$$path; git archive --format=tar --prefix=$(PKG_VER)/$$path/ $${rev}) > submodule.tar; \
+	  tar -A -f $${TARFILE_TMP} submodule.tar; \
+	  rm submodule.tar; \
+	done; \
+	mv $(PKG_VER).tar{.tmp,}; \
+	rm -f $(PKG_VER).tar.xz; \
+	xz $(PKG_VER).tar 
+
+srpm: dist-snapshot
+	(srcdir=$$(pwd); tmpdir=$$(mktemp -d); \
+	 sed -e "s,^Version:.*,Version: $(GITREV_FOR_PKG)," $(PACKAGE).spec > $${tmpdir}/$(PACKAGE).spec; \
+	 cp $(PKG_VER).tar.xz $${tmpdir}; \
+	 cd $${tmpdir} && ${srcdir}/rpmbuild-cwd -bs $(PACKAGE).spec)
```

Comment:
Makefiles are not my thing, but...

Should the resulting `atomic-pkglayer*src.rpm` get copied back to `$srcdir`?

When I ran `make -f Makefile.dist-packaging rpm`, I saw the following:

```
 cd ${tmpdir} && /home/miabbott/workspaces/atomic-pkglayer/rpmbuild-cwd -bs atomic-pkglayer.spec)
Wrote: /tmp/tmp.zYhUubnT0H/atomic-pkglayer-2015.3.10.g3935111-2.fc23.src.rpm
/home/miabbott/workspaces/atomic-pkglayer/rpmbuild-cwd --rebuild atomic-pkglayer-2015.3.10.g3935111*.src.rpm
error: cannot open atomic-pkglayer-2015.3.10.g3935111*.src.rpm: No such file or directory
Makefile.dist-packaging:35: recipe for target 'rpm' failed
make: *** [rpm] Error 1
```